ddl/ddl_api.go

```diff
@@ -2305,6 +2338,75 @@ func (d *ddl) AddColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTab
 	return errors.Trace(err)
 }
 
+// AddColumns will add multi new columns to the table.
+func (d *ddl) AddColumns(ctx sessionctx.Context, ti ast.Ident, specs []*ast.AlterTableSpec) error {
+	schema, t, err := d.getSchemaAndTableByIdent(ctx, ti)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	var columns []*table.Column
+	var positions []*ast.ColumnPosition
+	var offsets []int
```

Comment:
move line #2347 ~ 2349 after line #2366, and pre-allocation memory for those arrays.